server/src/main/java/org/apache/druid/client/JsonParserIterator.java

```diff
@@ -146,6 +158,16 @@ private void init()
           objectCodec = jp.getCodec();
         }
       }
+      catch (TimeoutException ignored) {
+        throw new QueryInterruptedException(
+            new ResourceLimitExceededException(
```

Comment:
I think `TimeoutException` is more appropriate than `ResourceLimitExceededException` here. It's supposed to be used for only physical resources such as memory.